instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,294 @@
+                    start_date=start_date
+                )
+            ))
+            sys.exit(1)
+
+        self.collect_instance_statistics(out, options['domain'], start_date, end_date)
+
+    def get_instance_from_domain_name(self, domain_name):
+        """ Get an instance object for a given domain name """
+        try:
+            instance = OpenEdXInstance.objects.get(
+                Q(external_lms_domain=domain_name) | Q(internal_lms_domain=domain_name)
+            )
+        except OpenEdXInstance.DoesNotExist:
+            self.stderr.write(self.style.ERROR(
+                'No OpenEdXInstance exists with an external or internal '
+                'domain of {domain_name}'.format(
+                    domain_name=domain_name
+                )
+            ))
+            sys.exit(1)
+
+        # If there are no active appservers for the instance, we should error out
+        if not instance.successfully_provisioned or not instance.get_active_appservers():
+            self.stderr.write(self.style.ERROR(
+                'No active OpenEdXAppServers exist for the instance with '
+                'external or internal domain of {domain_name}'.format(
+                    domain_name=domain_name
+                )
+            ))
+            sys.exit(1)
+
+        return instance
+
+    def get_elasticsearch_statistics(self, playbook_output_dir, name_prefix, start_date, end_date):
+        """ Execute the collect_elasticsearch_data playbook to gather statistics """
+        inventory = '[apps]\n{server}'.format(server=settings.INSTANCE_LOGS_SERVER_SSH_URL)
+        playbook_path = os.path.join(
+            settings.SITE_ROOT,
+            'playbooks/collect_instance_statistics/collect_elasticsearch_data.yml'
+        )
+
+        def log_line(line):
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_elasticsearch_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(
+                os.path.dirname(playbook_path),
+                'requirements.txt'
+            ),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'start_date: {start_date}\n'
+                'end_date: {end_date}'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix,
+                start_date=start_date,
+                end_date=end_date
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_LOGS_SERVER_SSH_USERNAME,
```

Comment:
@pcockwell, this has to be updated to match whatever the updated variable name you use in `settings.py`.

Reply:
Done in next commit